app/code/Magento/CatalogGraphQl/etc/schema.graphqls

```diff
@@ -10,6 +10,10 @@ type Query {
         sort: ProductAttributeSortInput @doc(description: "Specifies which attributes to sort on, and whether to return the results in ascending or descending order.")
     ): Products
     @resolver(class: "Magento\\CatalogGraphQl\\Model\\Resolver\\Products") @doc(description: "The products query searches for products that match the criteria specified in the search and filter attributes.") @cache(cacheIdentity: "Magento\\CatalogGraphQl\\Model\\Resolver\\Product\\Identity")
+    product (
```

Comment:
Seconding this - input should be a non-nullable list type of IDs.

Fun fact: GraphQL spec allows clients to pass a single value when the input is a list type, and it will coerce to a list length of 1 with that value:

![image](https://user-images.githubusercontent.com/5233399/84697298-86db9b80-af13-11ea-9e6d-57fcccdfb924.png)

Reply:
One other note: if we accept a list type as an input, we'll need a different name for this field that isn't singular.